### PR TITLE
Tree handler refactor

### PR DESCRIPTION
**Pull request by kRichard32**
Refactored the tree to use polymorphism instead of switch statements. As well, separated all the individual cases from the various tree handlers, into their own classes. As well, organized the tree code into directories. As well, various tests for the classes in the tree, and commenting

### CODE REVIEW

File: src/main/java/TreeHandlers/MainTreeNodeLogic/Roll.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                        CornerTilePerformActionUseCase cornerTileInteractor = new CornerTilePerformActionInteractor();
+                        assert landedOnCell instanceof CornerTiles;
+                        CornerTiles cornerTile = (CornerTiles) landedOnCell;
+
+                        setAnswer(cornerTileInteractor.performAction(currentPlayer, cornerTile));
+                        break;
+                    case "Action Space":
+                        PerformActionSpaceUseCase actionSpaceInteractor = new PerformActionSpaceCardInteractor();
+                        ActionSpace actionSpace = (ActionSpace) landedOnCell;
+                        setAnswer(actionSpaceInteractor.performAction(actionSpace, currentPlayer, board));
```
Comment (harshkpatel) [inline src/main/java/TreeHandlers/MainTreeNodeLogic/Roll.java:70]: I will also look into turning this into polymorphism

File: src/test/java/TreeHandlers/AuctionNodeLogic/AuctionTreeNodeLogicTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    public void testCheckAuction(){
+
+    }
+
+    @Test
+    public void testInitialize(){
+
+    }
+
+}
```
Comment (harshkpatel) [inline src/test/java/TreeHandlers/AuctionNodeLogic/AuctionTreeNodeLogicTest.java:27]: make sure to finish writing these tests!

**Review by harshkpatel — APPROVED** [review]
Great Refactor overall! I'm glad you took care of most of the switch statements and it understandable why you left out the two that still exist because that was not really part of your use case; I'll see what I can do about those. The tests look good coverage wise but some still need to be implemented. It's good that you laid out what you are going to test though despite the fact they are empty. No additional small comments, the structure of the code looks good and there are like 1-2 functions which can be considered too big, but after giving them a second look, it look somewhat necessary to keep them that way, so I do not have any issues with that. I will be adding the tests to the issues section and add @derrcodes as he is the one finishing up all the tests. If he can get them done before the PR is merged, that would be great, otherwise create a new PR!

**Review by benxu3 — APPROVED** [review]
Great work!! Thank you for refactoring the Tree Handlers to comply with SOLID and fixing the previous switch statement code smell. I will implement the creation of a loaded game in LoadGame.java, CreateLoadedGame.java, and UseCaseInteractor.java as soon as we merge in this PR. That way we will be able to write the tests for InitialNodeLogic. Again, incredible work!

**Review by derrcodes — APPROVED** [review]
Good job refactoring such a large amount of code so that it would adhere to clean architecture, code smells and SOLID. As for the other tests that Harsh mentioned those will be implemented soon. I had some issues testing some of the use cases but most of them are done. If I'm not mistaken Richard said he would look into those since I needed some help. I am also aware that due to the refactor, JavaDoc is incomplete and I will try to finish that as soon as possible.

**Pull request merged**